rir/src/utils/Set.h

```diff
@@ -60,6 +60,8 @@ class SmallSet {
     iterator end() { return container.end(); }
     const_iterator cbegin() const { return container.cbegin(); }
     const_iterator cend() const { return container.cend(); }
+    const_iterator begin() const { return container.cbegin(); }
+    const_iterator end() const { return container.cend(); }
```

Comment:
Why are these also constant iterators?

Reply:
tbh i don't fully understand the interplay of constant and iterators. suggestions?

Reply:
[Looking at the C++ API for vector iterators, this looks right](https://en.cppreference.com/w/cpp/container/vector/begin).

`begin()`/`end()` can be const or non-const, and it depends on the variable you're declaring. In the examples below, the compiler will pick the right version of `begin()`/`end()`:

```
for (const auto& b : bar) {
  // uses the const versions of begin()/end()
  // b is const, not allowed to modify b or bar
}
for (auto& b : bar) {
  // uses the non-const versions of begin()/end()
  // b is non-const, can modify it
}
```

But sometimes it's not clear:

```
for (auto it = bar.begin(); it != bar.end(); ++it) {
  // it is a pointer, and you can modify it and *it
}
for (const auto it = bar.begin(); it != bar.end(); ++it) {
  // this won't compile, because it is const so ++it is illegal!
}
for (const auto it = bar.cbegin(); it != bar.cend(); ++it) {
  // now this works
  // allowed to modify it, but not *it (or bar)
}
```

Reply:
ok, so cbegin/cend is just for when the compiler fails to pick the right one. thanks